test/core/libp2p/transport/tcp_transport_integration_test.cpp

```diff
@@ -0,0 +1,119 @@
+/**
+ * Copyright Soramitsu Co., Ltd. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <gtest/gtest.h>
+#include "libp2p/multi/multiaddress.hpp"
+#include "libp2p/transport/tcp.hpp"
+#include "testutil/outcome.hpp"
+
+using namespace libp2p::transport;
+using namespace libp2p::multi;
+using namespace boost;
+using kagome::common::Buffer;
+
+/**
+ * @given boost asio context, initialized transport and single listener
+ * @when create listener, assign callbacks
+ * @then no errors
+ * @when sync write methods executed
```

Comment:
look like at least two different cases

Reply:
wontfix.
test is pretty big, but every line is necessary.
Otherwise, it may result into many small tests (which are harder to maintain)